### PR TITLE
Changes regarding tests

### PR DESCRIPTION
**Pull request by pallabez**
### Issue:
Fixes - #170

### Description:
- add pinia/testing package
- update goal model
- add goal_mock fixture
- fix older tests
- add tests for store/goals add action

### Dev Tested:

- [x] Yes
- [ ] No

### Put the Feature behind Feature Flag

- [ ] Yes
- [x] No

### Test Stats
image

### Follow-up Issues (if any):

> NOTE: After your PR is merged in main please verify its working on prod and raise a dev to main PR

### CODE REVIEW

File: tests/integration/userAdapter.spec.ts [hunk 1]
```diff
@@ -1,10 +1,9 @@
-import { expect } from 'chai'
-import { fetchUsers } from '../../adapters/user'
-import { describe, test } from 'vitest'
+import * as userAdapter from '~/adapters/user'
+import { describe, test, expect } from 'vitest'
 
 describe('adapter/user', async () => {
-  test('response should be an array', async () => {
-    const users = await fetchUsers()
+  test.skip('response should be an array', async () => {
```
Comment (pallabez) [inline tests/integration/userAdapter.spec.ts:5]: Planning on fixing this test on a different PR.

File: tests/unit/store/goals.spec.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    beforeEach(() => {
+        const pinia = createPinia();
+        setActivePinia(pinia)
+    })
+
+    afterEach(() => {
+        vi.restoreAllMocks();
+    })
+
+    describe('add', async () => {
```
Comment (sahsisunny) [inline tests/unit/store/goals.spec.ts:18]: only add ?

File: tests/fixture/goal.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    description: 'MOCK_DESCRIPTION',
+    createdBy: 'MOCK_CREATED_BY',
+    startsOn: 'MOCK_STARTS_ON',
+    endsOn: 'MOCK_ENDS_ON',
+    percentageCompleted: 0,
+    assignedBy: 'MOCK_ASSIGNED_BY',
+    status: 'MOCK_STATUS',
+    assignedTo: 'MOCK_ASSIGNED_TO',
+
+}
```
Comment (sahsisunny) [inline tests/fixture/goal.ts:14]: The fixtures data should resemble the original data, rather than the current format.

**Comment by sahsisunny** [comment]
If possible, please include a screenshot of the test coverage.

File: tests/fixture/goal.ts (hunk 3 above)
Comment (sahsisunny) [reply tests/fixture/goal.ts:14]: and also attach the issue ticket

Comment (pallabez) [reply tests/fixture/goal.ts:14]: Sure

File: tests/unit/store/goals.spec.ts (hunk 2 above)
Comment (pallabez) [reply tests/unit/store/goals.spec.ts:18]: Its the function name.

**Review by sahsisunny — APPROVED** [review]

**Pull request merged**